gnuradio-runtime/include/gnuradio/math.h

```diff
@@ -31,6 +31,9 @@
 #include <gnuradio/api.h>
 #include <gnuradio/gr_complex.h>
 
+#define M_SQRT2	1.41421356237309504880	/* sqrt(2) */
```

Comment:
Seems safer to add `#ifndef M_SQRT2` here, in case we run into more non-standard behaviour elsewhere.

Reply:
Was thinking about that, but thought this shouldn't break, ever. My compiler didn't even complain about redefining this, when running with `-std=gnu(++)11`, or am I wrong?

Reply:
Let me quote Martin's law: "If you say it should, it won't". No harm in adding those lines.

Reply:
Hm, don't fully agree with here. If this **redefines** (as in: sets a _different_ value) `M_SQRT2`, I'd rather have *my* my √2 in my tests than some weird other people's not-quite-√2. If I was to feel paranoid about anything, it would be mathematical inconsistency, and then I'd favor @noc0lour's solution.

Reply:
notice that this doesn't express any opinion of me as a maintainer. I don't really feel worried about it, either way. I see that @mbr0wn's solution will probably save us a warning on `${strange platform}`, but that @noc0lour is closer to what one mathematically wants. So: `\shrug`.

Reply:
Our comments are orthogonal. You can #ifdef FOO -> #undef FOO, or you can #ifndef FOO ->#define FOO. In any case, you're adding a check for previous existence instead of just running `#define` and hoping for the best.

Note that `M_SQRT2` is a not-quite-but-almost-standard symbol. Things would be different if we were using our own symbols (`gnuradio::math::SQRT2` or something like that). Let's say you're redefining `M_PI`, which (I think) is always defined in math.h. Even if your definition is "better" than the original one, you could end up with mathematical inconsistencies between your code and other libs that use `M_PI` because you're redefining it in the middle of a source code flow.

Reply:
Ok then let's define `GR_M_*` to save us hassle to put in ifdef guards for every constant we might or might not define?

Reply:
Perfect.

Reply:
I was looking at other math defines e.g. `M_PI` which is used all over the place. It isn't defined in neither the c++11 nor the c11 standard. Yet it all works out?

Quick testing with `C++11` vs `C11` reveals that at least for `C++` I don't need to define `M_PI`. For `C` I do otherwise it doesn't know of `M_PI`. Both compiling with `-std=c{++}11`. I'm pretty sure I just have to include `<cmath>` here and be done, without redefining anything....

Reply:
My gut feeling tells me, there is still something wrong, I pushed changes with `GR_M_{sqrt2,log2e}`. But other constants `M_` remain and everything builds correctly with `cmake_c{xx}_extensions` turned off.